AuthenticationConfig RedisCachingProvider serialization error

## Description of problem
There is a caching error during login when running an installation using Azure Redis cache provider which expects all cache items to be serializable.  This is a DNN installation using the provider plugin available here: https://github.com/davidjrh/dnn.rediscachingprovider

DotNetNuke.Providers.RedisCachingProvider.RedisCachingProvider - Error while trying to store in cache the key DNN_Authentication.DNN_0 (Object type: DotNetNuke.Services.Authentication.AuthenticationConfig): System.Runtime.Serialization.SerializationException: Type 'Microsoft.Extensions.DependencyInjection.ServiceProvider' in Assembly 'Microsoft.Extensions.DependencyInjection, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' is not marked as serializable.

## Steps to reproduce

1. Ensure you have a functional DNN installation using Azure Redis Cache Provider 
2. Verify you're moving data into your cache via your Azure portal
2. Restart your DNN App Pool
3. Login
4. View the log4net file for the error as shown above


## Additional Context
It may be related to the AuthenticationConfigBase protected IServiceProvider DependencyProvider property.  Initial thought was to test adding the NonSerialized attribute but that won't work with protected members.  This error is showing in an App Service install but it is likely that it affects any installation that tries to implement the Redis Caching Provider plugin or Redis in general.  It's possible this is an installation specific issue.  This is the only DNN error I see when reviewing the logs for redis related issues.

## Affected browser
<!-- 
  Check all that apply, and add more if necessary. As appropriate, please specify the exact version(s) of the browser and operating system.
-->
* [x] Chrome
* [x] Firefox
* [x] Safari
* [x] Internet Explorer
* [x] Edge

@bdukes, @mitchelsellers  I put a note on the PR, Looking at the source in the main repo, I don't see the "[NonSerialized]" in the AuthenticationBase class, in fact, it looks totally different.

Also, I was looking closer at my logs and I'm seeing this:
`2021-08-19 15:09:52,438 [XXXXXXXXXXXX][Thread:6][ERROR] DotNetNuke.Providers.RedisCachingProvider.RedisCachingProvider - Error while trying to store in cache the key DNN_Authentication.DNN_1 (Object type: DotNetNuke.Services.Authentication.AuthenticationConfig): System.Runtime.Serialization.SerializationException: Type 'DotNetNuke.Web.Common.LazyServiceProvider' in Assembly 'DotNetNuke.Web, Version=9.10.0.0, Culture=neutral, PublicKeyToken=null' is not marked as serializable.`

Is the answer here as simple as decorating DotNetNuke.Web.Common.LazyServiceProvider
with [Serializable]?

The better question is why do I only see this on my production system and not on any of my test systems?

Hmm, interesting, Decorated the class to be serializable and deployed the dll to production, it seems to have reduced the number of occurrences by a great deal.

I have this one left:
`2021-08-23 01:30:43,188 [RD281878CE13C1][Thread:5][ERROR] DotNetNuke.Providers.RedisCachingProvider.RedisCachingProvider - Error while trying to store in cache the key DNN_Authentication.DNN_0 (Object type: DotNetNuke.Services.Authentication.AuthenticationConfig): System.Runtime.Serialization.SerializationException: Type 'Microsoft.Extensions.DependencyInjection.ServiceProvider' in Assembly 'Microsoft.Extensions.DependencyInjection, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' is not marked as serializable.
`

This one is MUCH less frequent, perhaps 200 times over the course of a day (we have a very busy site, so, its really not too bad).
I think this last one is never going to get resolved.  I'm going to make a change to the Redis caching provider and submit a PR to change the serialization errors to be warnings so that they can be quieted down for production.  but still seen on DEV systems.

I'm not seeing the original one you found.  Also, when I look @ the current DNN9.10.0 source, I do not see the decoration that you removed in your PR. (thus I'm confused and probably missing something important).

Yes, both my test and prod systems are running the Redis caching provider.

I feel like I'm playing whack-a-mole with this, I'm going to see if I still have some of the older logs that I was looking at to see how many different variations there were.

The really annoying thing is that I'm not sure where this is actually coming from.  I haven't had a great deal of success recreating it on demand.

--- a/DNN Platform/Library/Services/Authentication/AuthenticationConfigBase.cs
+++ b/DNN Platform/Library/Services/Authentication/AuthenticationConfigBase.cs
@@ -34,7 +34,6 @@ namespace DotNetNuke.Services.Authentication
         public int PortalID { get; set; }
 
         /// <summary>Gets the Dependency Provider to resolve registered services with the container.</summary>
-        [NonSerialized]
         protected IServiceProvider DependencyProvider => Globals.DependencyProvider;
     }
 }

--- a/DNN Platform/Library/Services/Authentication/AuthenticationConfigBase.cs
+++ b/DNN Platform/Library/Services/Authentication/AuthenticationConfigBase.cs
@@ -17,18 +17,13 @@ namespace DotNetNuke.Services.Authentication
     [Serializable]
     public abstract class AuthenticationConfigBase
     {
-        /// <summary>
-        /// Initializes a new instance of the <see cref="AuthenticationConfigBase"/> class.
-        /// </summary>
+        /// <summary>Initializes a new instance of the <see cref="AuthenticationConfigBase"/> class.</summary>
         public AuthenticationConfigBase()
         {
-            this.DependencyProvider = Globals.DependencyProvider;
         }
 
-        /// <summary>
-        /// Initializes a new instance of the <see cref="AuthenticationConfigBase"/> class.
-        /// </summary>
-        /// <param name="portalID"></param>
+        /// <summary>Initializes a new instance of the <see cref="AuthenticationConfigBase"/> class.</summary>
+        /// <param name="portalID">The portal ID.</param>
         protected AuthenticationConfigBase(int portalID)
             : this()
         {
@@ -38,13 +33,8 @@ namespace DotNetNuke.Services.Authentication
         [Browsable(false)]
         public int PortalID { get; set; }
 
-        /// <summary>
-        /// Gets the Dependency Provider to resolve registered
-        /// services with the container.
-        /// </summary>
-        /// <value>
-        /// The Dependency Service.
-        /// </value>
-        protected IServiceProvider DependencyProvider { get; }
+        /// <summary>Gets the Dependency Provider to resolve registered services with the container.</summary>
+        [NonSerialized]
+        protected IServiceProvider DependencyProvider => Globals.DependencyProvider;
     }
 }

